pytensor/tensor/math.py

```diff
@@ -597,10 +400,22 @@ def max_and_argmax(a, axis=None, keepdims=False):
     # Check axis and convert it to a Python list of integers.
     # Axis will be used as an op param of MaxAndArgmax.
     a = as_tensor_variable(a)
+
+    flag = False
```

Comment:
Code looks complicated, can we simplify? 

If we need to keep it, flag is not a readable name